doc/source/release/1.26.0-notes.rst

```diff
@@ -4,9 +4,64 @@
 NumPy 1.26.0 Release Notes
 ==========================
 
-The NumPy 1.26.0 release is a continuation of the 1.25.x release cycle, but
-with the distutils based build replaced by meson in order to work with Python
-3.12.
+The NumPy 1.26.0 release is a continuation of the 1.25.x release cycle with the
+addition of Python 3.12.0 support. Python 3.12 dropped distutils, consequently
+supporting it required finding a replacement for the setup/distutils based
+build system NumPy was using. We have chosen to use the Meson build system
+instead, and this is the first NumPy release supporting it. We have also moved
+to using the newly released Cython 3.0+. Supporting those two upgrades was a
```

Comment:
We still support the most recent `0.29.x` releases. I suggest rephrasing this as "this is also the first release that supports Cython 3.0, in addition to retaining 0.29.X compatibility".

The constraint in `pyproject.toml` should be `"Cython>=0.29.34,<3.1"`, just like in `pyproject.toml.setuppy`. I commented on this somewhere, but the change snuck in in a backport I think.

Also, the `requires-python` field in `pyproject.toml` should be changed to `requires-python = ">=3.9,<3.13".